### PR TITLE
Add Bats ubi8 utility image

### PR DESCRIPTION
**Pull request by pabrahamsson**
#### What is this PR About?
Adds a utility image containing [Bats](/bats-core/bats-core) and also
- Helm
- jq
- Openshift Client
- yq

The image can be used standalone to run bats test scenarios
```
podman run --rm -v $PWD:/code ubi8-bats /code/unit
```
It can also be used as part of a CI pipeline.

>**NOTE:** The required Quay repo has already been created: https://quay.io/repository/redhat-cop/ubi8-bats

#### How do we test this?
See included GH workflows

cc: @redhat-cop/day-in-the-life

### CODE REVIEW

File: .github/workflows/ubi8-bats-pr.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      - name: Build image
+        uses: docker/build-push-action@v1
+        with:
+          path: ${{ env.context }}
+          push: false
+          repository: ${{ env.image_name }}
+          tags: ${{ steps.check_version.outputs.IMAGE_TAGS }}
+      - name: Test image
+        run: |
+          echo "Running: docker run --entrypoint bash ${image_name}:${{ steps.check_version.outputs.IMAGE_TAGS }} -c 'bats --version'"
```
Comment (garethahealy) [inline .github/workflows/ubi8-bats-pr.yaml:28]: @pabrahamsson ; can we not use bats to do all these tests?

File: utilities/ubi8-bats/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,28 @@
+FROM registry.access.redhat.com/ubi8/ubi-minimal
```
Comment (garethahealy) [inline utilities/ubi8-bats/Dockerfile:1]: should we start using the .io registry? needs auth, not sure if that's been thought about when doing a build via gh-actions

File: .github/workflows/ubi8-bats-pr.yaml (hunk 1 above)
Comment (pabrahamsson) [reply .github/workflows/ubi8-bats-pr.yaml]: We sure can, I've updated the PR

File: utilities/ubi8-bats/Dockerfile (hunk 2 above)
Comment (pabrahamsson) [reply utilities/ubi8-bats/Dockerfile:1]: Are there benefits using `registry.redhat.io` over `registry.access.redhat.com`?

Comment (JaredBurck) [reply utilities/ubi8-bats/Dockerfile:1]: `registry.access.redhat.com` will be decommissioned sometime in the coming year. It is recommended that we prepare to use `registry.redhat.io`.  However, that registry does require authentication and supports registry tokens.

https://access.redhat.com/RegistryAuthentication

Comment (pabrahamsson) [reply utilities/ubi8-bats/Dockerfile:1]: Thanks @JaredBurck I think I knew that was coming but didn't have a timeline.
It looks as though the Docker GH action that we use supports auth for pulling and thus it should be possible to use `registry.redhat.io`.
@etsauer @sabre1041 thoughts on changing registry?

**Comment by garethahealy** [comment]
merging. can raise another PR if we decide to move base images to io registry.

**Pull request merged**